scripts/deploy/src/install.js

```diff
@@ -0,0 +1,163 @@
+            throw new Error("certificate_crt_file_path and certificate_key_file_path must be set in values when cert_source is 'specify-file-path'");
+        }
+        fs.copyFileSync(crtFilePath, 'certificate.crt');
+        fs.copyFileSync(keyFilePath, 'private.key');
+    } else if (certSource === 'my-ip-co') {
+        const crtResponse = await axios.get('https://local-ip.medicmobile.org/fullchain');
+        const keyResponse = await axios.get('https://local-ip.medicmobile.org/key');
+        fs.writeFileSync('certificate.crt', crtResponse.data);
+        fs.writeFileSync('private.key', keyResponse.data);
+    } else if (certSource !== 'eks-medic') {
+        throw new UserRuntimeError("cert_source must be either 'specify-file-path', 'my-ip-co', or 'eks-medic'");
+    }
+}
+
+function createSecret(namespace, values) {
+    console.log("Checking if secret api-tls-secret already exists...");
+    try {
+        execSync(`kubectl get secret api-tls-secret -n ${namespace}`, { stdio: 'inherit' });
+    } catch (err) {
+        console.log("Secret does not exist. Creating Secret from certificate and key...");
+        obtainCertificateAndKey(values);
+        execSync(`kubectl -n ${namespace} create secret tls api-tls-secret --cert=certificate.crt --key=private.key`, { stdio: 'inherit' });
+        fs.unlinkSync('certificate.crt');
+        fs.unlinkSync('private.key');
+    }
+}
+
+async function get_image_tag(chtversion) {
+    const response = await axios.get(`https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:${chtversion}`);
+    const data = response.data;
+    const tag = data.tags.find(tag => tag.container_name === 'cht-api');
+    if (!tag) throw new Error('cht image tag not found');
+    return tag.image.split(':').pop();
+}
+
+function helmInstallOrUpdate(valuesFile, namespace, values, image_tag) {
+    const chart_version = values.cht_chart_version || DEFAULT_CHART_VERSION;
+    ensureMedicHelmRepo();
+    const project_name = values.project_name || "";
+    try {
+        const releaseExists = execSync(`helm list -n ${namespace}`).toString();
+        if (releaseExists.includes(project_name)) {
+            console.log("Release exists. Performing upgrade.");
+            execSync(
+                `helm upgrade --install ${project_name} ${CHT_CHART_NAME} --version ${chart_version} --namespace ${namespace} --values ${valuesFile} --set cht_image_tag=${image_tag}`, { stdio: 'inherit' });
+            console.log(`Instance at ${values.ingress.host} upgraded successfully.`);
+        } else {
+            console.log("Release does not exist. Performing install.");
+            execSync(`helm install ${project_name} ${CHT_CHART_NAME} --version ${chart_version} --namespace ${namespace} --values ${valuesFile} --set cht_image_tag=${image_tag}`, { stdio: 'inherit' });
+            console.log(`Instance at ${values.ingress.host} installed successfully.`);
+        }
+    } catch (err) {
+        console.error(err.message);
```

Comment:
Applies to all error handling:
I find it a bit inconsistent, I would prefer to exit on any error (`process.exit(1);` as in other places).
Would it make sense to move all error handling to a general error handler instead of having try catch in many places?